source/includes/_transactions.md

```diff
@@ -77,18 +82,32 @@ message Signature {
 }
 ```
 
-**Payload** stores everything except signatures: 
+**Payload** stores all transaction fields, except signatures: 
 <ul>
     <li> Time of creation (unix time, in milliseconds) </li> 
-    <li> ID of transaction creator (username@domain) </li>
-    <li> Transaction counter — counts how many times transaction creator send transactions in total. Counter is used to prevent replays, and is formed on client side. </li>
-    <li> Repeated commands which would be described in details later</li> 
+    <li> Account ID of transaction creator (username@domain) </li>
+    <li> Transaction counter. It counts how many times transaction creator sent transactions in total. Counter is used to prevent replay attack, and is formed on client side </li>
+    <li> Repeated commands which are described in details in <a href="#commands">commands section</a> </li> 
 </ul>
 
-**Signatures** contain one or many signatures (ed25519 pubkey + signature):
+**Signatures** contain one or many signatures (ed25519 public key + signature):
 
 <aside class="notice">
-In preview version, there is no way to get current transaction counter by any means. You have to remember it on a client side and increment it. In future version this will be changed, and client application may retrieve this information.
+In the current version, there is no way to get current transaction counter by any means. You have to remember it on a client side and increment it. In a future version, this will be changed, and client application may retrieve this information.
 </aside>
 
 ## Transaction statuses
+
+The current version of Iroha peer follows client [pull principle](https://en.wikipedia.org/wiki/Pull_technology) for networking. It means that client should be proactive and request the state of a transaction from a peer. 
+
+This section describes the set of states and matches them with transaction lifecycle.
+
+![Block](../images/tx_status.png "Block structure")
+
+ * `NOT_RECEIVED`: the peer does not this transaction.
+ * `STATELESS_VALIDATION_FAILED`: the transaction was formed with some fields, not meeting constraints. This status is returned to a client, who formed transaction, right after the transaction was sent. 
+ * `STATELESS_VALIDATION_SUCCESS`: the transaction has successfully passed stateless validation. This status is returned to a client, who formed transaction, right after the transaction was sent. 
+ * `STATEFUL_VALIDATION_FAILED`: the transaction has commands, which violate validation rules, checking state of the chain (e.g. asset balance, account permissions, etc.) 
+ * `STATEFUL_VALIDATION_SUCCESS`: the transaction has successfully passed stateful validation.
+ * `IN_PROGRESS`: the transaction is the part of a block, which is on voting during current consensus round.
```

Comment:
If I am not mistaken, IN_PROGRESS means that transaction was received, but not processed.